src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/index.tsx

```diff
@@ -0,0 +1,409 @@
+import TextField from 'app/views/settings/components/forms/textField';
+import recreateRoute from 'app/utils/recreateRoute';
+import space from 'app/styles/space';
+import withApi from 'app/utils/withApi';
+
+import RuleNodeList from './ruleNodeList';
+
+const FREQUENCY_CHOICES = [
+  ['5', t('5 minutes')],
+  ['10', t('10 minutes')],
+  ['30', t('30 minutes')],
+  ['60', t('60 minutes')],
+  ['180', t('3 hours')],
+  ['720', t('12 hours')],
+  ['1440', t('24 hours')],
+  ['10080', t('one week')],
+  ['43200', t('30 days')],
+];
+
+const ACTION_MATCH_CHOICES = [['all', t('all')], ['any', t('any')], ['none', t('none')]];
+
+// TODO(ts): I can't get this to work if I'm specific -- should be: 'condition' | 'action';
+type ConditionOrAction = string;
+
+type Props = {
+  api: Client;
+  actions: IssueAlertRuleActionTemplate[] | null;
+  conditions: IssueAlertRuleConditionTemplate[] | null;
+  project: Project;
+  organization: Organization;
+} & RouteComponentProps<{orgId: string; projectId: string; ruleId: string}, {}>;
+
+type State = {
+  rule: IssueAlertRule | null;
+  loading: boolean;
+  error: null | {
+    [key: string]: string[];
+  };
+  environments: Environment[];
+};
+
+class IssueRuleEditor extends React.Component<Props, State> {
+  state: State = {
+    rule: null,
+    loading: false,
+    error: null,
+    environments: [],
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  async fetchData() {
+    const {
+      api,
+      params: {ruleId, projectId, orgId},
+    } = this.props;
+
+    const defaultRule = {
+      actionMatch: 'all',
+      actions: [],
+      conditions: [],
+      name: '',
+      frequency: 30,
+      environment: ALL_ENVIRONMENTS_KEY,
+    };
+
+    const promises = [
+      api.requestPromise(`/projects/${orgId}/${projectId}/environments/`),
+      ruleId
+        ? api.requestPromise(`/projects/${orgId}/${projectId}/rules/${ruleId}/`)
+        : Promise.resolve(defaultRule),
+    ];
+
+    try {
+      const [environments, rule] = await Promise.all(promises);
+      this.setState({environments, rule});
+    } catch (_err) {
+      addErrorMessage(t('Unable to fetch data'));
+    }
+  }
+
+  handleSubmit = async () => {
+    const {rule} = this.state;
+    const isNew = !rule || !rule.id;
+    const {project, organization} = this.props;
+
+    const endpoint = `/projects/${organization.slug}/${project.slug}/rules/${
+      rule && rule.id ? `${rule.id}/` : ''
+    }`;
+
+    if (rule && rule.environment === ALL_ENVIRONMENTS_KEY) {
+      delete rule.environment;
+    }
+
+    addLoadingMessage(t('Saving...'));
+
+    try {
+      const resp = await this.props.api.requestPromise(endpoint, {
+        method: isNew ? 'POST' : 'PUT',
+        data: rule,
+      });
+
+      this.setState({error: null, loading: false, rule: resp});
+
+      // Redirect to correct ID if new
+      if (isNew) {
+        browserHistory.replace(
+          recreateRoute(`${resp.id}/`, {...this.props, stepBack: -1})
```

Comment:
@billyvg might need `recreateRoute(`issue-rules/${resp.id}/`, {...this.props, stepBack: -1})` here.

Reply:
@billyvg also may need to apply the same change to https://github.com/getsentry/sentry/blob/a9be062057511c423448f36e210eafebb4f08603/src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx#L128

Reply:
Good catch, I've actually changed it to redirect back to the list view after edit/create since that's commonly what we do.